scripts/ci/selective_ci_checks.sh

```diff
@@ -121,76 +182,93 @@ function show_changed_files() {
 # Output:
 #    Count of changed files matching the patterns
 function count_changed_files() {
+    local count_changed_files
     count_changed_files=$(echo "${CHANGED_FILES}" | grep -c -E "$(get_regexp_from_patterns)" || true)
     echo "${count_changed_files}"
 }
 
-function run_all_tests_when_push_or_schedule() {
-    if [[ ${GITHUB_EVENT_NAME} == "push" || ${GITHUB_EVENT_NAME} == "schedule" ]]; then
-        echo
-        echo "Always run all tests in case of push/schedule events"
-        echo
-        set_outputs_run_all_tests
-        exit
+function check_if_api_tests_should_be_run() {
+    local pattern_array=(
+        "^airflow/api"
+    )
+    show_changed_files
```

Comment:
Wait, this works? `show_changed_files` calls `get_regexp_from_patterns`, and that references `pattern_array` -- and it can pick up the local array defined here?

That was not how I thought local worked. I can see how the scoping rules of bash made this the case, but I guess I was expecting something else.

Reply:
Yep. See here: https://tldp.org/LDP/abs/html/localvar.html. This is built in bash:

> However, as Thomas Braunberger points out, a local variable declared in a function is also visible to functions called by the parent function.